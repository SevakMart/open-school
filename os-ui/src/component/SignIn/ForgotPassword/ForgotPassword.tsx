import React, { useEffect, useState } from 'react';
import { validateEmail } from '../../../helpers/EmailValidate';
import authService from '../../../services/authService';
import ResetPassword from '../ResetPassword/ResetPassword';
import {
  EMAIL, FORGOT_PASSWORD, ENTER_EMAIL_FOR_VERIFICATION, SEND_CODE_NOTIFICATION,
  CONTINUE,
} from '../../../constants/Strings';
import CloseIcon from '../../../icons/Close';
import styles from './ForgotPassword.module.scss';

const ForgotPassword = (
  { returnToSignInForm }: {returnToSignInForm: ()=> void},
) => {
  const [email, setEmail] = useState('');
  const [successMessage, setSuccessMessage] = useState('');
  const [emailError, setEmailError] = useState('');
  const [resetPassword, setResetPassword] = useState(false);
  const { mainContainer, errorMessage, successedMessage } = styles;

  const handleEmailChange = (e:React.SyntheticEvent) => {
    setEmail((e.target as HTMLInputElement).value);
  };

  const sendForgotPassword = () => {
    const emailError = (validateEmail(email));
    if (!emailError) {
      authService.sendForgotPasswordRequest({ email })
        .then((response) => {
          if (response.status === 200) {
            setSuccessMessage(response.data.message);
            setEmailError('');
          } else if (response.status === 400) {
            setEmailError(response.data.message);
          }
        });
    } else setEmailError(emailError);
  };
  useEffect(() => {
    let timer:any;
    if (successMessage) {
      timer = setTimeout(() => setResetPassword(true), 3500);
    }
    return () => clearTimeout(timer);
  }, [successMessage]);

  return (
    <>
      {
        resetPassword ? (
          <ResetPassword
            returnToSignInForm={() => returnToSignInForm()}
            email={email}
          />
        )
          : (
            <>
              {
              successMessage ? <h3 data-testid="forgotPasswordSuccessMessage" className={successedMessage}>{successMessage}</h3>
                : (
                  <>
                    <CloseIcon handleClosing={() => returnToSignInForm()} />
                    <div className={mainContainer}>
                      <h2>{FORGOT_PASSWORD}</h2>
                      <p>{ENTER_EMAIL_FOR_VERIFICATION}</p>
                      <p>{SEND_CODE_NOTIFICATION}</p>
                      <div>
                        <label htmlFor="email">
                          {EMAIL}
                          <span style={{ color: 'red' }}> *</span>
                        </label>
                        <input
                          data-testid="forgotPasswordEmailInput"
                          id="email"
                          type="email"
                          name="email"
                          placeholder="ex: namesurname@gmail.com"
                          onChange={handleEmailChange}
                          required
                        />
                      </div>
                      {emailError ? (<h4 data-testid="emailErrorForgotPassword" className={errorMessage}>{emailError}</h4>) : null}
                      <button data-testid="continueButton" type="button" onClick={sendForgotPassword}>{CONTINUE}</button>
                    </div>
                  </>
                )
                }
            </>
          )
}
    </>
  );
};
export default ForgotPassword;
