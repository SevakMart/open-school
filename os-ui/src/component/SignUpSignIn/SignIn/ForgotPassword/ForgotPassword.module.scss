@import "../../../../styles/Mixins/ButtonMixin.scss";
@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.mainContainer {
    @include flexStyle(column, flex-start, space-between);
    margin-top: 3%;
    margin-bottom: 1%;
    padding-left: 8%;
    padding-right: 8%;
    padding-top: 4%;
    padding-bottom: 1%;

    h2{
        @include fontStyle("Montserrat", normal, 800, 2.25rem, 2.75rem);
        color: #222427;
        align-self:center;
    }

    div {
      position: relative;
      width: 100%;
      margin-top: 5%;
      @include flexStyle(column, flex-start, space-between);
    }
    input {
      width: 100%;
      border: 1px solid #d9dbe9;
      box-sizing: border-box;
      border-radius: 0.75rem;
      height: 3.5rem;
      margin-top: 2%;
      margin-bottom: 3%;
      padding-left: 3%;
    }
    label {
      @include fontStyle("Montserrat", normal, 500, 0.875rem, 1rem);
      color: #333941;
    }
    p {
      display: block;
      width: 100%;
      text-align: center;
      @include fontStyle("Montserrat", normal, 500, 1rem, 0.0rem);
      color: #333941;
      cursor: pointer;
    }
    button {
      @include buttonStyle(#5e617b,#ffffff,1rem,100%,3.5rem,none,none,none,none);
      @include fontStyle("Montserrat", normal, 500, 1rem, 1.25rem);
      display: block;
      text-align: center;
      border: 1px solid #5e617b;
      box-sizing: border-box;
      margin-top: 4%;
    }
  }

  .successedMessage {
    @include fontStyle("Montserrat", normal, 700, 1.25rem, 2.5rem);
    margin-top: 5%;
    margin-left: 5%;
    margin-right: 5%;
    color: #222427;
    text-align: center
  }

  .errorMessage {
    @include fontStyle("Montserrat", normal, 800, 0.875rem, 0rem);
    color: #ff0000;
  }

