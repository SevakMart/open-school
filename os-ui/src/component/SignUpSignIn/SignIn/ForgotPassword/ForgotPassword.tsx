import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { validateEmail } from '../../../../helpers/EmailValidate';
import authService from '../../../../services/authService';
import ResetPassword from '../ResetPassword/ResetPassword';
import CloseIcon from '../../../../icons/Close';
import styles from './ForgotPassword.module.scss';

const ForgotPassword = (
  { returnToSignInForm }: {returnToSignInForm: ()=> void},
) => {
  const { t } = useTranslation();
  const [email, setEmail] = useState('');
  const [successMessage, setSuccessMessage] = useState('');
  const [emailError, setEmailError] = useState('');
  const [resetPassword, setResetPassword] = useState(false);
  const { mainContainer, errorMessage, successedMessage } = styles;

  const handleEmailChange = (e:React.SyntheticEvent) => {
    setEmail((e.target as HTMLInputElement).value);
  };

  const sendForgotPassword = () => {
    const emailError = (validateEmail(email));
    if (!emailError) {
      authService.sendForgotPasswordRequest({ email })
        .then((response) => {
          if (response.status === 200) {
            setSuccessMessage(response.data.message);
            setEmailError('');
          } else if (response.status === 400) {
            setEmailError(response.data.message);
          }
        });
    } else setEmailError(emailError);
  };
  useEffect(() => {
    let timer:any;
    if (successMessage) {
      timer = setTimeout(() => setResetPassword(true), 3500);
    }
    return () => clearTimeout(timer);
  }, [successMessage]);

  return (
    <>
      {
        resetPassword ? (
          <ResetPassword
            returnToSignInForm={() => returnToSignInForm()}
            email={email}
          />
        )
          : (
            <>
              {
              successMessage ? <h3 data-testid="forgotPasswordSuccessMessage" className={successedMessage}>{successMessage}</h3>
                : (
                  <>
                    <CloseIcon />
                    <div className={mainContainer}>
                      <h2>{t('string.forgotPsd.title')}</h2>
                      <p>{t('string.forgotPsd.enterEmail')}</p>
                      <p>{t('string.forgotPsd.sendNotification')}</p>
                      <div>
                        <label htmlFor="email">
                          {t('form.labels.email')}
                          <span style={{ color: 'red' }}> *</span>
                        </label>
                        <input
                          data-testid="forgotPasswordEmailInput"
                          id="email"
                          type="email"
                          name="email"
                          placeholder={t('form.placeholder.email')}
                          onChange={handleEmailChange}
                          required
                        />
                      </div>
                      {emailError ? (<h4 data-testid="emailErrorForgotPassword" className={errorMessage}>{emailError}</h4>) : null}
                      <button data-testid="continueButton" type="button" onClick={sendForgotPassword}>{t('button.forgotPsd.continue')}</button>
                    </div>
                  </>
                )
                }
            </>
          )
}
    </>
  );
};
export default ForgotPassword;
