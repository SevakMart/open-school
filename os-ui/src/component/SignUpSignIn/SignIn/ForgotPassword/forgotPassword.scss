@import './../../../../styles/Mixins/ButtonMixin.scss';
@import './../../../../styles/Mixins/FontMixin.scss';
@import './../../../../styles/Configs/Fonts.scss';
@import './../../../../styles/Configs/Colors.scss';

.closeIconX {
  position: absolute;
  top: 24px;
  right: 24px;
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  cursor: pointer;
  &:hover {
    color: brown;
  }  
}

.forgotpsd_box {
  font-family: $primary-font;
  margin-top: 35px;
  &_title {
    text-align: center;
    font-weight: 800;
    font-size: 2.25rem;
    color:$header-main-color;
  }
  &_subtitles {
   margin-top: 20px; 
  }
  &_subtitle {
    font-size: 1rem;
    color:$main-text-color;
    margin-top: 10px;
  }
}

.ForgotPasswordBtns {
  display: flex;
  justify-content: flex-start;
  gap: 20px;
  margin-top: 40px;
}

.anyBtn {
  padding: 20px 0;
  width: 100%;
  font-family: $primary-font;
  font-weight: 500;
  font-size: 18px;
  border-radius: 0.7rem;
  background-color: $secondary-color;
  color: $white-text-color;
  display: block;
  text-align: center;
  box-sizing: border-box;
  border: none;
  margin-bottom: 24px;
}

.backButton {
  order: 1;
  width: 100%;
  margin-right: auto;
}

.continueButton {
  order: 2;
}

.anyBtn,
.backButton {
  &:hover {
    background-color: #83f0dc;
  }
  &:active {
    background-color: #17ac90;
  }
  &:hover[disabled] {
    background-color: $btn-primary-color;
    color: #fff;
  }

  &[disabled] {
    opacity: 0.5;
    cursor: not-allowed;
  }
}

.forgotpsdContainer {
  margin: 20px 80px 0 80px;
  &_mainBody {
    margin-top: 30px;    
  }
}

@media screen and (max-width: 1080px) {
  .anyBtn {
    font-size: 14px;
    padding: 15px 10px;
  }
}

@media screen and (max-width: 900px) {
  .ForgotPasswordBtns {
    display: block;
  }
}