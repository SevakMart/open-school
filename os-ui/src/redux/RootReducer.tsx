import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
