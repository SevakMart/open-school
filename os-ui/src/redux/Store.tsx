import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
