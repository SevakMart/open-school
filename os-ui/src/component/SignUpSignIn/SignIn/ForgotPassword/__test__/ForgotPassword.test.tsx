import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import ForgotPassword from '../ForgotPassword';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test cases for ForgotPassword component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ForgotPassword />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
