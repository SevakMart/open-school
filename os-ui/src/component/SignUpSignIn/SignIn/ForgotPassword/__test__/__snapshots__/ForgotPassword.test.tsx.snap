// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for ForgotPassword component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    data-testid="closeButton"
    style="background-color: transparent; color: rgb(132, 138, 157); cursor: pointer; float: inline-end; margin-right: 3%; font-size: 2rem; padding: 0px;"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7,7 L17,17 M7,17 L17,7"
        fill="none"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
  </div>
  <div
    class="mainContainer"
  >
    <h2>
      string.forgotPsd.title
    </h2>
    <p>
      string.forgotPsd.enterEmail
    </p>
    <p>
      string.forgotPsd.sendNotification
    </p>
    <div>
      <label
        for="email"
      >
        form.labels.email
        <span
          style="color: red;"
        >
           *
        </span>
      </label>
      <input
        data-testid="forgotPasswordEmailInput"
        id="email"
        name="email"
        placeholder="form.placeholder.email"
        required=""
        type="email"
      />
    </div>
    <button
      data-testid="continueButton"
      type="button"
    >
      button.forgotPsd.continue
    </button>
  </div>
</DocumentFragment>
`;
